Directory.Build.props

```diff
@@ -481,4 +481,9 @@
   <!-- Use Roslyn Compilers to build -->
   <Import Project="$(RoslynPropsFile)" Condition="'$(RunningOnUnix)'!='true' and Exists('$(RoslynPropsFile)') and '$(UseRoslynCompilers)'!='false' and '$(RoslynIncompatibleMsbuildVersion)' != 'true'" />
   <Import Project="$(RoslynPropsFile)" Condition="'$(RunningOnUnix)'=='true' and Exists('$(RoslynPropsFile)') and '$(RoslynIncompatibleMsbuildVersion)' != 'true'" />
+
+  <!-- Additional optimizations -->
+  <PropertyGroup>
+    <ILLinkClearInitLocals Condition="'$(IsTestProject)' != 'true' AND '$(Language)' != 'VB'">true</ILLinkClearInitLocals>
```

Comment:
Do we want to enable this for reference assemblies as well? Or should we just condition it to IsSourceProject == true?

What exactly does ILLinkClearInitLocals do, first time I see it :smile:?

Reply:
It doesn't really matter, but we could limit it to just src.  It tells the linker to remove a flag output by the C# compiler that would force the JIT to zero-out stack space used for things like stackalloc.  So without this, when you do `Span<char> s = stackalloc char[1000];`, the JIT needs to generate code that zeros out those 1000 bytes, whereas with the flag cleared, no such clearing is performed.

Reply:
Thanks for explaining. Given that, it doesn't matter if it is enabled for reference assemblies since we will never initialize stack space in there.

Reply:
It matters for the build performance. Invoking an extra process for each reference assembly is not free.

Reply:
Good point.  I'll add the constraint.

Reply:
I believe you want to use `IsRuntimeAssembly`. Note that this isn't defined in Directory.Build.targets so you need to move that over.

Reply:
IsRuntimeAssembly instead of IsSourceProject?  What do each of those mean and why is the latter insufficient?

Reply:
Forget it, `IsSourceProject` is fine. `IsRuntimeAssembly` is an implementation detail of our binplacing strategy.